### PR TITLE
MSSQL Source

### PR DESCRIPTION
**Pull request by cgardens**
## What
* MSSQL Source
* Trying to get this done for someone who has been asking for it on slack and wants to try it out this weekend: https://airbytehq.slack.com/archives/C01AHCDHCKE/p1605282114024300
* Had to rework the `JdbcAbstractSource` to handle the fact that jooq free version doesn't handle mssql.
* There a couple hacks in here that I'll call out. Up to the reviewer on whether it's appropriate to include them to get this done for the person who is asking or whether I should wait to fix them. I'm comfortable with them for the short term.

### CODE REVIEW

File: airbyte-integrations/connectors/source-mssql/src/main/resources/spec.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+  "documentationUrl": "https://docs.airbyte.io/integrations/destinations/mysql",
```
Comment (michel-tricot) [inline airbyte-integrations/connectors/source-mssql/src/main/resources/spec.json:2]: mssql

**Review by michel-tricot — APPROVED** [review]

File: airbyte-integrations/connectors/source-mssql/src/main/resources/spec.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  "connectionSpecification": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "MySql Source Spec",
```
Comment (michel-tricot) [inline airbyte-integrations/connectors/source-mssql/src/main/resources/spec.json:5]: here as well

File: airbyte-integrations/connectors/source-mssql/src/main/resources/spec.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      "host": {
+        "description": "Hostname of the database.",
+        "type": "string"
+      },
+      "port": {
+        "description": "Port of the database.",
+        "type": "integer",
+        "minimum": 0,
+        "maximum": 65536,
+        "default": 3306,
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-mssql/src/main/resources/spec.json:19]: is the UI ok with defaults now?

**Review by sherifnada — COMMENTED** [review]
This may be better if the methods being overridden by MSSQL source are abstract. Then we have a `AbstractJooqJdbcSource` and MSSQL source also extending. This would make things easier the next time we impl a source that JOOQ isn't easy to use with.

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java [hunk 4]
```diff
@@ -103,23 +103,36 @@ public AirbyteConnectionStatus check(JsonNode config) {
     }
   }
 
+  protected String getCurrentDatabaseName(DSLContext ctx) {
+    return ctx.select(currentSchema()).fetch().get(0).getValue(0, String.class);
```
[the remark this answers is not in the document]
Comment (cgardens) [reply airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java:107]: in both cases we were just using current schema to pull the db name (well technically in one case were just making sure it executed).

File: airbyte-integrations/connectors/source-mssql/src/main/resources/spec.json (hunk 3 above)
Comment (cgardens) [reply airbyte-integrations/connectors/source-mssql/src/main/resources/spec.json]: it should. artem pushed a fix. but i might remove it for now just to be safe.

**Comment by cgardens** [comment]
> This may be better if the methods being overridden by MSSQL source are abstract. Then we have a AbstractJooqJdbcSource and MSSQL source also extending. This would make things easier the next time we impl a source that JOOQ isn't easy to use with.

Fair. I'll try to come and finesse this.

**Comment by cgardens** [comment]
Ran a manual test in the UI to confirm that it worked as expected. Only tested against a locally run docker container version of mssql.

**Pull request merged**